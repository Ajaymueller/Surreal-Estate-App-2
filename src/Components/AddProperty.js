import React, { useState } from "react";
import Alert from "./Alert";
import PropertyForm from "./PropertyForm";
import { saveProperty } from "../requests";

import "../styles/AddProperty.css";

const initialState = {
  fields: {
    title: "",
    city: "Manchester",
    type: "",
    bedrooms: 0,
    bathrooms: 0,
    price: 0,
    email: "",
  },
  alert: {
    message: "",
    isSuccess: false,
  },
};

const AddProperty = () => {
  const [fields, setFields] = useState(initialState.fields);
  const [alert, setAlert] = useState(initialState.alert);

  const handleAddProperty = (event) => {
    event.preventDefault();
    setAlert({ message: "", isSuccess: false });

    return saveProperty(fields, setAlert);
  };

  const handleFieldChange = (event) => {
    const changedField = event.target.name;
    const newValue = event.target.value;

    setFields({ ...fields, [changedField]: newValue });
  };

  return (
    <div className="add-property" data-testid="addProperty">
       <PropertyForm 
        handleAddProperty={handleAddProperty}
        handleFieldChange={handleFieldChange}
        title={fields.title}
        city={fields.city}
        type={fields.type}
        bedrooms={fields.bedrooms}
        bathrooms={fields.bathrooms}
        price={fields.price}
        email={fields.email}
        />
      <Alert message={alert.message} success={alert.isSuccess} />
    </div>
  );
};

export default AddProperty;