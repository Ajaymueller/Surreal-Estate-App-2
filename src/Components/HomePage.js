import React from "react";
import "../styles/HomePage.css";

const HomePage = () => {
    return (
      <div className="home-page">
        <aside className="home-page__welcome">
          <h2 className="home-page__heading">Make your dream home a reality</h2>
          <p>
            Find your Dream Home here with Surreal Estate 
          </p>
        </aside>
      </div>
    );
  };

  export default HomePage;