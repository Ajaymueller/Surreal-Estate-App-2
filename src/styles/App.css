@import url("https://fonts.googleapis.com/css2?family=Noto+Sans+JP:wght@300&display=swap");

.App {
  padding: 10px;
  display: flex;;
  grid-gap: 10px;
  flex-direction: column;
  flex-wrap: wrap;
  min-height: 100vh;
  font-family: 'Noto Sans JP', sans-serif;
}