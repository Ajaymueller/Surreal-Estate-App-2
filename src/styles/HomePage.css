.home-page {
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    background-image: url("images/houses-background.jpeg");
    background-repeat: no-repeat;
    background-size: cover;
    height: 100vh;
}

.home-page h2 {
    font-size: 100px;
    text-align: center;
}

.home-page p {
    font-size: xx-large;
    text-align: center;
}