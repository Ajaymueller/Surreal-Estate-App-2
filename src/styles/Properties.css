.Properties {
    margin-left: 160px;
    padding: 12px;
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
}