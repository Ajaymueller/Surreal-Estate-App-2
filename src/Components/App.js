import React, { useState } from "react";
import { Switch, Route, BrowserRouter } from "react-router-dom";
import "../styles/App.css";

import NavBar from "./NavBar";
import Properties from "./Properties";
import AddProperty from "./AddProperty";
import Favourites from "./Favourites";
import HomePage from "./HomePage";
import Footer from "./Footer";

const App = () => {
  const [userID, setUserID] = useState("");

  const handleLogin = (response) => {
    setUserID(response.userID);
  };

  const handleLogout = () => window.FB.logout(() => setUserID(""));

  return (
    <div className="App">
      <NavBar onLogin={handleLogin} onLogout={handleLogout} userID={userID} />
      <BrowserRouter>
      <Switch>
      <Route
          path="/"
          exact
          render={(props) => <HomePage {...props} userID={userID} />}
        />
        <Route
          path="/properties"
          exact
          render={(props) => <Properties {...props} userID={userID} />}
        />
        <Route
          path="/saved-properties"
          exact
          render={(props) => <Favourites {...props} userID={userID} />}
        />
        <Route path="/add-property" component={AddProperty} exact />
      </Switch>
      </BrowserRouter>
      <Footer />
    </div>
  );
};

export default App;
