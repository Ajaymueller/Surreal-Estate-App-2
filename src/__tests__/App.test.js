import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import App from '../Components/App';
import Properties from '../Components/Properties';
import AddProperty from '../Components/AddProperty'; 
import { MemoryRouter } from 'react-router-dom';
import { Route } from 'react-router-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HomePage from '../Components/HomePage';
//import { faItalic } from '@fortawesome/free-solid-svg-icons';
Enzyme.configure({ adapter: new Adapter() });

jest.mock("react-facebook-login", () =>
  jest.fn(() => <div>Facebook login</div>)
);

describe("App", () => {
    it("renders correctly", () => {
    const { asFragment } = render(
    <MemoryRouter><App /></MemoryRouter> )
    expect(asFragment).toMatchSnapshot();
    });
});

describe("With React router", () => {
    xit("renders Properties and AddProperty component", () => {
    const { getByText, getByTestId } = render(
    <MemoryRouter><App /></MemoryRouter> )
    fireEvent.click(getByText("View Properties"));
    fireEvent.click(getByText("Add Property"));
    expect(getByTestId("properties")).toBeInTheDocument();
    expect(getByTestId("addProperty")).toBeInTheDocument();
    });
});

describe("routes using memory router", () => {
    xit("should load HomePage component for / route", () => {
    const component = mount ( <MemoryRouter initialentries="{['/']}">
    <App/> </MemoryRouter>)
    expect(component.find(HomePage)).toHaveLength(1);
    })
    xit("should show properties component for /properties route", () => {
    const component = mount ( <MemoryRouter initialentries="{['/properties']}">
    <App/> </MemoryRouter> )
    expect(component.find(Properties)).toHaveLength(1);
    });
    xit("should show AddProperty component for /add-property", () => {
    const component = mount ( <MemoryRouter initialentries="{['/add-property']}">
    <App/> </MemoryRouter> )
    expect(component.find(AddProperty)).toHaveLength(1);
    })
})