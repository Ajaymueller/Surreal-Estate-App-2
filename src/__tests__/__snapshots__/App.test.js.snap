exports[`App renders correctly 1`] = `[Function]`;
